### DISCUSSION TITLE
multi_cell triggering page break even though not at the bottom of the page

### DISCUSSION THREAD

**Discussion by joshua4788** [py-pdf/fpdf2#931] [Q&A]
Hi there, 

Hope this post finds you well.

I have been racking my head to fix this issue but am out of my depths here, so will greatly appreciate it anybody could shed a light.

I have applied auto page break by setting pdf.set_auto_page_break(auto=True, margin=15). My PDF document consists of two columns of multi_cells for which the same x and y positions are set for each row of multi_cells. Multi_cells are populated with values populated from my Python Dash web page.

The auto page break works perfectly fine in some occasions, but in other occasions, page break is triggered consecutively for multiple multi_cells even though they are not reaching the bottom of the page.

For example, my code has the following multi_cells (each pair of multi_cell_left and multi_cell_right forms a row):
- multi_cell_left_1
- multi_cell_right_1
- multi_cell_left_2
- multi_cell_right_2
- multi_cell_left_3
- multi_cell_right_3

If multi_cell_left_2 and multi_cell_right_2 causes an auto page break because they are at the end of the page, multi_cell_left_3 gets printed on the next page (page 2) and multi_cell_right_3 gets printed on the following page (page 3).

This has been driving me crazy, so if there is anyone who has faced this issue before and had a fix, I will greatly appreciate it.

If you need a minimal code, please let me know and I will upload.

Thank you very much.,

**Comment by andersonhc** [comment]
Hi @joshua4788 

Thank you for reporting this bug.

> If you need a minimal code, please let me know and I will upload.

If you can provide the minimal code it would help debugging this problem.

**Comment by andersonhc** [comment]
By the way, I believe your code would be simpler with a table instead of side-by-side multi_cells

Take a look on the [documentation about tables](https://py-pdf.github.io/fpdf2/Tables.html) and check our [test files with tables](https://github.com/py-pdf/fpdf2/blob/7f4e0a39d0a3f0b6a7a5bcee0993a16ed5abcaab/test/table/test_table.py#L129C4-L129C4), many of them use multiline.
